columnphysics/icepack_therm_itd.F90

```diff
@@ -1021,215 +1015,142 @@ subroutine lateral_melt (dt,         ncat,       &
-
-         do n = 1, ncat
+      do n = 1, ncat
 
       !-----------------------------------------------------------------
       ! Melt the ice and increment fluxes.
       !-----------------------------------------------------------------
 
-            ! fluxes to coupler
-            ! dfresh > 0, dfsalt > 0, dfpond > 0
-
-            dfresh = (rhoi*vicen(n) + rhos*vsnon(n))      * rsiden(n) / dt
-            if (saltflux_option == 'prognostic') then
-               sicen = c0
-               do k=1,nilyr
-                  sicen = sicen + trcrn(nt_sice+k-1,n) / real(nilyr,kind=dbl_kind)
-               enddo
-               dfsalt = rhoi*vicen(n)*sicen*p001 * rsiden(n) / dt
-            else
-               dfsalt = rhoi*vicen(n)*ice_ref_salinity*p001 * rsiden(n) / dt
-            endif
-            fresh  = fresh + dfresh
-            fsalt  = fsalt + dfsalt
-
-            if (tr_pond_topo) then
-               dfpond = aicen(n)*trcrn(nt_apnd,n)*trcrn(nt_hpnd,n)*rsiden(n)
-               fpond  = fpond - dfpond
-            endif
+         ! fluxes to coupler
+         ! dfresh > 0, dfsalt > 0, dfpond > 0
 
-            ! history diagnostics
-            meltl = meltl + vicen(n)*rsiden(n)
+         dfresh = (rhoi*vicen(n) + rhos*vsnon(n))      * rsiden(n) / dt
+         if (saltflux_option == 'prognostic') then
+            sicen = c0
+            do k=1,nilyr
+               sicen = sicen + trcrn(nt_sice+k-1,n) / real(nilyr,kind=dbl_kind)
+            enddo
+            dfsalt = rhoi*vicen(n)*sicen*p001 * rsiden(n) / dt
+         else
+            dfsalt = rhoi*vicen(n)*ice_ref_salinity*p001 * rsiden(n) / dt
+         endif
+         fresh  = fresh + dfresh
+         fsalt  = fsalt + dfsalt
 
-            ! state variables
-            vicen_init(n) = vicen(n)
-            aicen(n) = aicen(n) * (c1 - rsiden(n))
-            vicen(n) = vicen(n) * (c1 - rsiden(n))
-            vsnon(n) = vsnon(n) * (c1 - rsiden(n))
+         if (tr_pond_topo) then
+            dfpond = aicen(n)*trcrn(nt_apnd,n)*trcrn(nt_hpnd,n)*rsiden(n)
+            fpond  = fpond - dfpond
+         endif
 
-            ! floe size distribution
-            if (tr_fsd) then
-               if (rsiden(n) > puny) then
-                  if (aicen(n) > puny) then
+         ! history diagnostics
+         meltl = meltl + vicen_init(n)*rsiden(n)
 
-                     ! adaptive subtimestep
-                     elapsed_t = c0
-                     afsd_tmp(:) = afsdn_init(:,n)
-                     d_afsd_tmp(:) = c0
-                     nsubt = 0
+         ! state variables
+         aicen(n) = aicen(n) * (c1 - rsiden(n))
+         vicen(n) = vicen(n) * (c1 - rsiden(n))
+         vsnon(n) = vsnon(n) * (c1 - rsiden(n))
 
-                     DO WHILE (elapsed_t.lt.dt)
+         ! floe size distribution
+         if (tr_fsd) then
+            if (rsiden(n) > puny) then
+               if (aicen(n) > puny) then    ! not sure if this should be aicen or aicen_init
```

Comment:
What's the plan for figuring out if it should be aicen or aicen_init?  Why not use the most recent value?  I guess the vertical melting and lateral melting are proceeding simultaneously, so neither is exactly correct.  (aicen+aicen_init)/2?

Reply:
Actually, is aicen_init at the beginning of the timestep or some other point during the timestep?  Seems to be set at the beginning of the lateral_melt subroutine.

Reply:
I would think this should be aicen_init, which isn't defined yet.  Otherwise aicen might be less than puny from the state variable adjustment immediately above, and the fsd wouldn't be updated equivalently.

Reply:
When I look at this code, it seems like the if conditions on rsiden and aicen are kind of redundant. Except in the case where rsiden(n) = 1. This would mean that aicen(n) would go to zero. So, I do think you want a check on the updated aicen(n).